fibers.scm

```diff
@@ -120,11 +120,21 @@
       (unless init
         (error "run-fibers requires initial fiber thunk when creating sched"))
       (spawn-fiber (lambda ()
-                     (call-with-values init
-                       (lambda vals (atomic-box-set! ret vals)))
-                     ;; Could be that this fiber was migrated away.
-                     ;; Make sure to wake up the main scheduler.
-                     (spawn-fiber (lambda () #t) scheduler))
+                     (dynamic-wind
+		       (const #t)
+		       (lambda ()
+			 (catch #t
```

Comment:
Don't use `catch`, as it destroys all backtrace information.  Instead use `with-exception-handler` which doesn't destroy it (maybe with a fallback to 'catch' for old versions of Guile if it's not Guile 2.2).  If you go for `with-exception-handler`, also go for `raise-continuable` instead of `raise-exception`/`raise`, otherwise you accumulate `raise-exception` in the backtrace, which is harmless but noisy.